todo/src/store/currentUser/userSlice.ts

```diff
@@ -0,0 +1,21 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { RootState } from "../../store";
+export const userSlice = createSlice({
+  name: "user",
+  initialState: { user: null ,userName:null, loggedIn:false},
+
+  reducers: {
+    login: (state, action) => {
+      state.user = action.payload.data;
+      state.userName=action.payload.email;
+      state.loggedIn= true ;
+    },
+    signOut:(state,action)=>{
+      state.loggedIn=false;
```

Comment:
why not clear the entire state and set it to the initial state?

Reply:
I have made changes and pushed it.